.github/workflows/codeql.yml

```
@@ -0,0 +1,72 @@
+name: "Swift CodeQL Analysis"
+
+on:
+  workflow_dispatch:
+  push:
+    branches: [ "dev" ]
+  pull_request:
+    branches: [ "dev" ]
+      
+jobs:
+
+  analyze:
+    name: Analyze Swift
+    runs-on: macos-latest
+    defaults:
+      run:
+        working-directory: ./client/ios
+    permissions:
+      security-events: write
+      packages: read
+      actions: read
+      contents: read
+
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+        - language: swift
+          build-mode: manual
+       
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v4
+
+      # Setup swift version 6.0
+    - name: Setup Swift
+      uses: swift-actions/setup-swift@v2
+      with:
+        swift-version: '6.0'
```

maybe use 6.1 since this is the latest version
```suggestion
        swift-version: '6.1'
```